library/ecdh.c

```diff
@@ -94,61 +101,195 @@ void mbedtls_ecdh_init( mbedtls_ecdh_context *ctx )
     memset( ctx, 0, sizeof( mbedtls_ecdh_context ) );
 }
 
+static int mbedtls_ecdh_setup_internal( mbedtls_ecdh_context *ctx,
+                                        mbedtls_ecp_group_id grp )
+{
+    int ret;
+#if defined(MBEDTLS_ECDH_LEGACY_CONTEXT)
+    mbedtls_ecdh_context *real_ctx = ctx;
+#else
+    mbedtls_ecdh_context_mbed *real_ctx;
+
+    if( ctx->ctx.mbed_ecdh != NULL )
```

Comment:
I think `mbed_ecdh` should be the structure, not a pointer, which would simplify this function.